internal/compress/lz4_test.go

```diff
@@ -171,39 +139,162 @@ func Test_lz4Compressor_CompressVector(t *testing.T) {
+			args: args{
+				vector: []float32{0.1, 0.2, 0.3},
+			},
+			fields: fields{
+				gobc: &MockCompressor{
+					CompressVectorFunc: func(vector []float32) (bytes []byte, err error) {
+						return nil, errors.New("compress err")
+					},
+				},
+				compressionLevel: 0,
+				lz4: &lz4.MockLZ4{
+					NewWriterLevelFunc: func(w io.Writer, level int) lz4.Writer {
+						return &lz4.MockWriter{
+							WriteFunc: w.Write,
+							FlushFunc: func() error {
+								return nil
+							},
+							CloseFunc: func() error {
+								return nil
+							},
+						}
+					},
+				},
+			},
+			want: want{
+				want: nil,
+				err:  errors.New("compress err"),
+			},
+		},
+		{
+			name: "returns (nil, error) when Write fails",
+			args: args{
+				vector: []float32{0.1, 0.2, 0.3},
+			},
+			fields: fields{
+				gobc: &MockCompressor{
+					CompressVectorFunc: func(vector []float32) (bytes []byte, err error) {
+						return []byte("vdaas/vald"), nil
+					},
+				},
+				compressionLevel: 0,
+				lz4: &lz4.MockLZ4{
+					NewWriterLevelFunc: func(w io.Writer, level int) lz4.Writer {
+						return &lz4.MockWriter{
+							WriteFunc: func(p []byte) (int, error) {
+								return 0, errors.New("write err")
+							},
+							FlushFunc: func() error {
+								return nil
+							},
+							CloseFunc: func() error {
+								return nil
+							},
+						}
+					},
+				},
+			},
+			want: want{
+				want: nil,
+				err:  errors.New("write err"),
+			},
+		},
+		{
+			name: "returns (nil, error) when Flush fails",
+			args: args{
+				vector: []float32{0.1, 0.2, 0.3},
+			},
+			fields: fields{
+				gobc: &MockCompressor{
+					CompressVectorFunc: func(vector []float32) (bytes []byte, err error) {
+						return []byte("vdaas/vald"), nil
+					},
+				},
+				compressionLevel: 0,
+				lz4: &lz4.MockLZ4{
+					NewWriterLevelFunc: func(w io.Writer, level int) lz4.Writer {
+						return &lz4.MockWriter{
+							WriteFunc: w.Write,
+							FlushFunc: func() error {
+								return errors.New("flush err")
+							},
+							CloseFunc: func() error {
+								return nil
+							},
+						}
+					},
+				},
+			},
+			want: want{
+				want: nil,
+				err:  errors.New("flush err"),
+			},
+		},
+		{
+			name: "returns (nil, error) when Close fails",
+			args: args{
+				vector: []float32{0.1, 0.2, 0.3},
+			},
+			fields: fields{
+				gobc: &MockCompressor{
+					CompressVectorFunc: func(vector []float32) (bytes []byte, err error) {
+						return []byte("vdaas/vald"), nil
+					},
+				},
+				compressionLevel: 0,
+				lz4: &lz4.MockLZ4{
+					NewWriterLevelFunc: func(w io.Writer, level int) lz4.Writer {
+						return &lz4.MockWriter{
+							WriteFunc: w.Write,
+							FlushFunc: func() error {
+								return nil
+							},
+							CloseFunc: func() error {
+								return errors.New("close err")
+							},
+						}
+					},
+				},
+			},
+			want: want{
+				want: nil,
+				err:  errors.New("close err"),
+			},
```

Comment:
Could you please add test case that occurs error when Write or Flush and Clouse returns error ?

Reply:
??? I already created.
what did you mean?

Reply:
sorry. :bow:
I could not find the following test case...

```go
{
			name: "returns (nil, error) when Close fails",
			args: args{
				vector: []float32{0.1, 0.2, 0.3},
			},
			fields: fields{
				gobc: &MockCompressor{
					CompressVectorFunc: func(vector []float32) (bytes []byte, err error) {
						return []byte("vdaas/vald"), nil
					},
				},
				compressionLevel: 0,
				lz4: &lz4.MockLZ4{
					NewWriterLevelFunc: func(w io.Writer, level int) lz4.Writer {
						return &lz4.MockWriter{
							WriteFunc: func(p []byte) (int, error) {
								return 0, errors.New("write err")
							},
							FlushFunc: func() error {
								return nil
							},
							CloseFunc: func() error {
								return errors.New("close err")
							},
						}
					},
				},
			},
			want: want{
				want: nil,
				err:  errors.Wrap(errors.New("write err"),errors.New("close err").Error())
			},
		},
```

but you have no need to fix it, as it has coverage.
I was making a mistake.:bow: